Physical_Engine/Modify/Transform.cs

```diff
@@ -0,0 +1,82 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.Collections.Generic;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using BH.oM.Physical.Elements;
+using System.Linq;
+using BH.oM.Physical.FramingProperties;
+using BH.Engine.Spatial;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Interface Methods - IElements             ****/
+        /***************************************************/
+
+        [Description("Transforms the IFramingElement's location and orientation angle by the transform matrix. Only rigid body transformations are supported.")]
+        [Input("framingElement", "IFramingElement to transform.")]
+        [Input("transform", "Transform matrix.")]
+        [Output("transformed", "Modified IFramingElement with unchanged properties, but transformed location and orientation angle.")]
+        public static IFramingElement Transform(this IFramingElement framingElement, TransformMatrix transform)
+        {
+            if (!transform.IsRigidTransformation())
+            {
+                BH.Engine.Reflection.Compute.RecordError("Transformation failed: only rigid body transformations are currently supported.");
+                return null;
+            }
+
+            IFramingElement result = framingElement.GetShallowClone() as IFramingElement;
+            result.Location = result.Location.ITransform(transform);
+
+            ConstantFramingProperty property = result.Property as ConstantFramingProperty;
+            if (property == null)
+                BH.Engine.Reflection.Compute.RecordWarning($"Orientation angle of the IFramingElement has not been transformed because its property is not ConstantFramingProperty. BHoM_Guid: {framingElement.BHoM_Guid}");
+            else
+            {
+                Line locationBefore = framingElement.Location as Line;
+                if (locationBefore == null)
+                    BH.Engine.Reflection.Compute.RecordWarning($"Orientation angle of the IFramingElement has not been transformed because the element is not linear. BHoM_Guid: {framingElement.BHoM_Guid}");
```

Comment:
It's probably my lack of experience with `IFramingElement` interface but why don't we transform the orientation angle of non-linear elements?

Reply:
I am not sure how would that be done tbh (not even sure how is orientation defined in case of nonlinear elements 🙈). Possibly I could do that by reproducing the linear approach on the line spanning between start and end of the location curve, but what if the curve is closed? Etc. etc.

I would rather not add anything than add something I am not too sure about. As long as anyone can tell me how to do that properly.

Reply:
Can maybe help out a bit here..

https://github.com/BHoM/documentation/wiki/BHoM-Structural-Conventions#coordinate-system gives the conventions.

Basically, for a planar non-linear element, the starting normal is defined by the plane of the curve, and then the infinite numer of normals along the curve is rotated around the curve axis with the orientation angle.

Unless I am completely mistaken, I think this mean that this actually gets simpler... Just transforming the curve itself, should basically bring the normal with it, meaning you are done there. Good if one of you guys can have a think about that as well just making sure I am not completely wrong.

For non-planar curves we actually do not have any convention yet as no usecases requiring it has popped up yet (as Revit does not allow them, it is far less common). Not 100% sure what to do about them yet either, but might for the purpose of this PR just transform the curves as well, as I am pretty sure any convention we put in will be dependant on the Location curve, alternatively, just skip non-planar curves for now and raise an error.

Reply:
Big thanks for help @IsakNaslundBh! The above is very informative and makes perfect sense (as confirmed during our discussion) 👍 I have updated the code the reflect the following convention:
- linear - same as structural bars
- nonlinear planar - only transform the curve, rotation being driven by its plane
- nonlinear nonplanar - only transform the curve + a warning that correctness could not be ensured 100%

I think this is maximum we can get for now. I have also updated UT and GH script, so the PR should be ready for re-review.